great_expectations/data_context/store/store.py

```diff
@@ -233,7 +233,18 @@ def get_all(self) -> list[Any]:
         if self.cloud_mode:
             objs = self.gx_cloud_response_json_to_object_collection(objs)
 
-        return list(map(self.deserialize, objs))
+        deserializable_objs: list[Any] = []
+        bad_objs: list[Any] = []
+        for obj in objs:
+            try:
+                deserializable_objs.append(self.deserialize(obj))
+            except Exception:  # We want to filter out all undeserializable configs
```

Comment:
i know that this is in the top level store, so deserialize will have an arbitrary implementation, but i think we want to actually just catch `ValidationError` here. Is there a known use case in which we need to catch a non-pydantic error?